java/017-subsets.md

```diff
@@ -4,9 +4,48 @@
 - [next: 018. Subsets II](018-subsets-ii.md)
 
 ---
+### Notes:
+- Iterative algorithm for Power set:
+![iterative algorithm](http://chart.googleapis.com/chart?cht=tx&chl=PS%5C(%5C%7Be%5C%7D%5Ccup%20A%5C)%3D%5C%7B%5C%7Be%5C%7D%5Ccup%20A'%7C%20%5Cforall%20A'%5Cin%20PS(A)%5C%7D%5Ccup%20PS(A))
+- recursive algorithm for Poswer Set:
+TODO
 
-put your own notes and solutions here.
-you can add any reference link such as [title](reference url) here.
+### Code:
+#### Iterative algorithm for Power set:
+```java
+class Solution {
+    /**
+     * @param S: A set of numbers.
+     * @return: A list of lists. All valid subsets.
+     */
+    public ArrayList<ArrayList<Integer>> subsets(int[] nums) {
+        //First sort the array nums in descending order (using bubble sort)
+        for(int i=0;i<nums.length-1;i++){
```

Comment:
Hello, java has a library function `Arrays.sort()`, which would be more convenience to sort an array:) For `Collections` such as `ArrayList`, java has `Collections.sort()` too.

Arrays：
https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html

Collections:
https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html